internal/controller/postgrescluster/patroni.go

```diff
@@ -87,14 +89,31 @@ func (r *Reconciler) handlePatroniRestarts(
 	// replicas here, replicas will typically restart first because we see them
 	// first.
 	if primaryNeedsRestart != nil {
+		pod := primaryNeedsRestart.Pods[0]
 		exec := patroni.Executor(func(
 			ctx context.Context, stdin io.Reader, stdout, stderr io.Writer, command ...string,
 		) error {
-			pod := primaryNeedsRestart.Pods[0]
 			return r.PodExec(ctx, pod.Namespace, pod.Name, container, stdin, stdout, stderr, command...)
 		})
 
-		return errors.WithStack(exec.RestartPendingMembers(ctx, "master", naming.PatroniScope(cluster)))
+		patroniVerStr, ok := cluster.Annotations[pNaming.ToCrunchyAnnotation(pNaming.AnnotationPatroniVersion)]
+		if !ok {
+			return errors.New("patroni version annotation was not found")
+		}
+		patroniVer, err := gover.NewVersion(patroniVerStr)
+		if err != nil {
+			return errors.Wrap(err, "failed to get patroni ver")
+		}
+		patroniVer4 := patroniVer.Compare(gover.Must(gover.NewVersion("4.0.0"))) >= 0
```

Comment:
i see we do this in multiple places, can we move this logic to a function?

Reply:
also i don't think we should use `gover.Must` since it'll panic if it can't parse the version

Reply:
https://github.com/percona/percona-postgresql-operator/pull/921/commits/e070313a3e004aabd4b89cebc5c6c86b4057fcdf